src/popover/popover.directive.ts

```diff
@@ -84,31 +95,172 @@ export class PopoverContainer {
 	 * Emits an event when the state of `isOpen` changes. Allows `isOpen` to be double bound
 	 */
 	@Output() isOpenChange = new EventEmitter<boolean>();
-
+	/**
+	 * Show caret for the
+	 */
 	@HostBinding("class.cds--popover--caret") @Input() caret = true;
+	/**
+	 * Enable drop shadow around the popover container
+	 */
 	@HostBinding("class.cds--popover--drop-shadow") @Input() dropShadow = true;
-	@HostBinding("class.cds--popover--high-contrast") @Input() highContrast = false;
-	@HostBinding("class.cds--popover--open") @Input() isOpen = false;
-
+	/**
+	 * Enable high contrast for popover container
+	 */
+	@HostBinding("class.cds--popover--high-contrast") @Input() highContrast = true;
+	/**
+	 * **Experimental**: Use floating-ui to position the tooltip
+	 */
+	@HostBinding("class.cds--popover--auto-align") @Input() autoAlign = false;
 	@HostBinding("class.cds--popover-container") containerClass = true;
-	@Input() align: "top" | "top-left" | "top-right" |
-		"bottom" | "bottom-left" | "bottom-right" |
-		"left" | "left-bottom" | "left-top" |
-		"right" | "right-bottom" | "right-top" = "bottom";
+	@HostBinding("class.cds--popover--open") _open = false;
+
+	protected popoverContentRef: HTMLElement;
+	protected caretRef: HTMLElement;
+	protected caretOffset;
+	protected caretHeight;
+
+	protected autoUpdateCleanUp: Function;
 
-	constructor(private changeDetectorRef: ChangeDetectorRef) {}
+	constructor(
+		protected elementRef: ElementRef,
+		protected ngZone: NgZone,
+		protected renderer: Renderer2,
+		protected changeDetectorRef: ChangeDetectorRef
+	) { }
 
-	handleChange(open: boolean, event: Event) {
-		if (this.isOpen !== open) {
+	/**
+	 * Handles emitting open/close event
+	 * @param open - Is the popover container open
+	 * @param event - Event
+	 */
+	handleChange(open: boolean, event?: Event) {
+		if (this._open !== open) {
 			this.isOpenChange.emit(open);
 		}
 
 		if (open) {
-			this.onOpen.emit(event);
+			if (event) {
+				this.onOpen.emit(event);
+			}
+
+			// auto alignment is enabled, we use auto update to set the placement for the element
+			if (this.autoAlign) {
+				if (this.caretRef) {
+					// Get caret offset/height property
+					this.caretOffset = parseFloat(
+						getComputedStyle(this.caretRef).getPropertyValue("--cds-popover-offset").split("rem", 1)[0]
+					) * 16 || 10;
+					this.caretHeight = parseFloat(
+						getComputedStyle(this.caretRef).getPropertyValue("--cds-popover-caret-height").split("px", 1)[0]
```

Comment:
`getComputedStyle` triggers a reflow (https://gist.github.com/paulirish/5d52fb081b3570c81e3a#calling-getcomputedstyle) which is expensive, I'd avoid doing it twice but cache the results in a variable

Reply:
Cool, didn't know about this, thanks for pointing it out! Will update soon.